helix/appendix/alternate-visual-studio-structures.rst

```diff
@@ -0,0 +1,231 @@
+                msbuild.config('overrideParams', overrideParams);
+                msbuild.config('version', config.buildToolsVersion);
+                msbuild.build();
+            });
+
+    You can alternatively use Sitecore TDS or Helix Publishing Pipeline
+    for local deployments, as seen in the new Sitecore Helix Examples.
+
+
+Are there other tooling improvements which can help?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+If your primary concern is the overhead of creating projects which follow
+the Sitecore Helix conventions (naming, filesystem structure, etc.), then
+you might consider some of the community tools which make this easier by
+providing project "templates" which save you some clicks. Take care when
+using these however that you remove any generated projects
+`which the module does not need <#does-your-solution-contain-projects-you-do-not-need>`__.
+
+To improve the performance of Visual Studio and reduce build times, you can
+also consider using the `Filtered Solutions <https://docs.microsoft.com/en-us/visualstudio/ide/filtered-solutions?view=vs-2019>`__
+feature of Visual Studio 2019 to reduce the number of projects open when working
+on a particular module.
+
+
+The role of Visual Studio projects
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Aside from the pragmatic issues, the architectural question at the root of
+this decision is the role of a project/assembly in Visual Studio. This is
+essentially the subject of Robert C. Martin's *Principles of Package and Component Design*,
+which contain the `underlying principles <package-principles>`__ that Sitecore Helix
+is based on. Based on these principles, Visual Studio projects form a "physical" boundary
+and unit of change for modules in Sitecore Helix.
+
+Martin however makes an assumption that modules (packages) are released / deployed
+independently. Due to the versioning and testing complexity this would put on the typical Sitecore
+solution, Sitecore Helix actually `discourages partial deploys of modules </devops/deployment/strategy>`__,
+meaning that the unit of release is the entirety of the application. If we are not deploying
+and versioning these assemblies independently, it is a valid question as to whether
+the modules really merit their own Visual Studio projects.
+
+The modular architecture practices of Sitecore Helix though are meant to improve solution
+*maintainability*. The convention of using a separate web application
+project for each module (which deploys to Sitecore CM/CD) means that all the code and
+configuration related to a module are located in one place. Combine that with the module
+filesystem and solution folder structure, and you also have the items, tests, and other related
+code for that module. Practically, it's very easy for a future developer to find/discover
+the code related to a desired system change. It's also easier to track and manage the dependencies
+to and from that module.
```

Comment:
imo this should drive home further that the goal is dependency management - that you literally cannot _use_ a module without referencing it. This prevents accidental dependencies that are undesirable. I've seen helix-ifying things like this tease out subtle referencing bugs - especially when template codegen is involved - because the codegen requires a code-level reference to pull in a template model.

Reply:
added a bit more emphasis to dependency management